src/Sulu/Bundle/AdminBundle/Admin/Navigation/NavigationExternalItem.php

```diff
@@ -0,0 +1,453 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Admin\Navigation;
+
+class NavigationExternalItem implements \Iterator, NavigationItemInterface
```

Comment:
This class is missing proper type information. We should use the php typehinting:

```php
/**
* @var string
*/
protected $id;
```

will become
```php
protected string $id;
```

That way we have help from php checking the types. Same with arguments and return types.